src/vconsole/vconsole-setup.c

```diff
@@ -328,48 +320,79 @@ static void setup_remaining_vcs(int fd, bool utf8) {
         }
 }
 
+static const char *find_source_vc(const char *vc, int *src_fd, int *src_idx) {
+        static char ttyname[sizeof("/dev/tty") - 1 + DECIMAL_STR_MAX(int)];
```

Comment:
humm, static variables like this I don't really like I must say. It's fine to use static variables if the data placed there is effectively constant, but doing this in a function that places different data there depending on the parameters is something I don#t like at all I must say.

Sure, this is a private function here, that is not exposed to anything else, but still, I think we really shouldn't even do this.

There's nothing wrong with allocating a string for this on the heap.

I'd much prefer if we'd change this to something like this:

```c
static int find_source_vc(char **ret_path, int *ret_idx);
```